packages/astro-rss/src/index.ts

```diff
@@ -1,113 +1,127 @@
 	/** (required) Description of the RSS Feed */
-	description: string;
+	description: z.infer<typeof rssOptionsValidator>['description'];
 	/**
 	 * Specify the base URL to use for RSS feed links.
-	 * We recommend "import.meta.env.SITE" to pull in the "site"
-	 * from your project's astro.config.
-	 */
-	site: string;
-	/**
-	 * List of RSS feed items to render. Accepts either:
-	 * a) list of RSSFeedItems
-	 * b) import.meta.glob result. You can only glob ".md" (or alternative extensions for markdown files like ".markdown") files within src/pages/ when using this method!
+	 * We recommend using the [endpoint context object](https://docs.astro.build/en/reference/api-reference/#contextsite),
+	 * which includes the `site` configured in your project's `astro.config.*`
 	 */
-	items: RSSFeedItem[] | GlobResult;
+	site: z.infer<typeof rssOptionsValidator>['site'];
+	/** List of RSS feed items to render. */
+	items: RSSFeedItem[];
 	/** Specify arbitrary metadata on opening <xml> tag */
-	xmlns?: Record<string, string>;
+	xmlns?: z.infer<typeof rssOptionsValidator>['xmlns'];
 	/**
 	 * Specifies a local custom XSL stylesheet. Ex. '/public/custom-feed.xsl'
 	 */
-	stylesheet?: string | boolean;
+	stylesheet?: z.infer<typeof rssOptionsValidator>['stylesheet'];
 	/** Specify custom data in opening of file */
-	customData?: string;
+	customData?: z.infer<typeof rssOptionsValidator>['customData'];
 	/** Whether to include drafts or not */
-	drafts?: boolean;
+	drafts?: z.infer<typeof rssOptionsValidator>['drafts'];
 };
 
 type RSSFeedItem = {
 	/** Link to item */
 	link: string;
+	/** Full content of the item. Should be valid HTML */
+	content?: string;
 	/** Title of item */
-	title: string;
+	title: z.infer<typeof rssSchema>['title'];
 	/** Publication date of item */
-	pubDate: Date;
+	pubDate: z.infer<typeof rssSchema>['pubDate'];
 	/** Item description */
-	description?: string;
-	/** Full content of the item, should be valid HTML */
-	content?: string;
+	description?: z.infer<typeof rssSchema>['description'];
 	/** Append some other XML-valid data to this item */
-	customData?: string;
+	customData?: z.infer<typeof rssSchema>['customData'];
 	/** Whether draft or not */
-	draft?: boolean;
+	draft?: z.infer<typeof rssSchema>['draft'];
 };
 
-type GenerateRSSArgs = {
-	rssOptions: RSSOptions;
-	items: RSSFeedItem[];
-};
-
-function isGlobResult(items: RSSOptions['items']): items is GlobResult {
-	return typeof items === 'object' && !items.length;
+type ValidatedRSSFeedItem = z.infer<typeof rssFeedItemValidator>;
+type ValidatedRSSOptions = z.infer<typeof rssOptionsValidator>;
+
+const rssFeedItemValidator = rssSchema.extend({ link: z.string(), content: z.string().optional() });
+const rssOptionsValidator = z.object({
+	title: z.string(),
+	description: z.string(),
+	site: z.string(),
+	items: z.array(rssFeedItemValidator),
+	xmlns: z.record(z.string()).optional(),
+	drafts: z.boolean().default(false),
```

Comment:
Typo?
```suggestion
	draft: z.boolean().default(false),
```

Reply:
Actually not! This mirrors the naming of `markdown.drafts`, aka "include drafts yes or no." I fought to rename to `includeDrafts` and lost 😅